tools/pythonpkg/requirements-dev.txt

```
@@ -4,9 +4,13 @@ setuptools_scm>=6.3
 setuptools>=58
 pytest
 pandas
-pyarrow
+pytest-timeout
 mypy
 psutil>=5.9.0
+requests>=2.26
 fsspec
-polars
+pyarrow>=8.0
+torch
+polars<0.20.22
```

```suggestion
polars
```
I think this is not necessary anymore since we skip on import failures.

---

I'm confused

The terminology here is too similar
The `optional_requirements.py` does not fail if the install of the module failed, and instead keeps going
the tests skip if the module is not available

What happened here is that the install didn't fail, and instead the `import polars` was causing a `Fatal Python Error: Illegal instruction`
Nothing we have set up will prevent that

---

I am also a bit confused by this all.
Question is: does limiting the polars version helps/makes sense, or that can be reverted to any polars version?

---

What we tried:
- polars 0.20.22
- polars 0.20.21
- disable polars
- disable tensorflow

I am not convinced that the problem lies with polars or tensorflow
Both of them caused a `Fatal Python Error: Illegal instruction` inside their `__init__.py` script (runs when `import <name>` is called)

But it seems suspicious to me that *only* those packages were problematic.
Our (heavy handed) fix for this is to disable both polars and tensorflow for the time being

The effect of this is that our code that interacts with these packages goes from *potentially* being tested on OSX to never being tested on OSX.

@carlopi

---

I say *potentially* because we were already silently ignoring the package install failure for these packages

---

A side effect of this PR is that on Linux we will stop testing with latest polars but testing only against 0.20.21. Not serious, but I think it would be cleaner to revert to no upper bound

Other point is whether we care or not about testing polars on OSX, to be discussed separately.